### PR TITLE
chore: Move Codecov, wasm-build, Bitrot & doc-tests to push

### PR DESCRIPTION
**Pull request by CPerezz**
This should cut down significantly the CI times on every push done to a branch for a PR.
Resolves: #124

### CODE REVIEW

**Comment by CPerezz** [comment]
Checking the current workflows triggered seems that the change was done correctly!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: CI checks
 
-on: [push, pull_request]
+on: [pull_request]
```
Comment (han0110) [inline .github/workflows/ci.yml:3]: Should we also run this when `push`? Since we don't require PR to be on the same base of `main` iirc.

Comment (CPerezz) [reply .github/workflows/ci.yml:3]: That's true!!!!

Comment (CPerezz) [reply .github/workflows/ci.yml]: Fixed in 9ddd4043

**Review by han0110 — APPROVED** [review]
LGTM!

**Pull request merged**